internal/controllers/machinedeployment/machinedeployment_controller.go

```diff
@@ -82,7 +82,6 @@ func (r *Reconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, opt
 
 	err = ctrl.NewControllerManagedBy(mgr).
 		For(&clusterv1.MachineDeployment{}).
```

Comment:
I took a closer look into the mapping functions. Looks like they don't enqueue at all based on ownerRefs

```
	// Check if the controller reference is already set and
	// return an empty result when one is found.
	for _, ref := range ms.ObjectMeta.GetOwnerReferences() {
		if ref.Controller != nil && *ref.Controller {
			return result
		}
	}
```

I think the goal of these function is to only enqueue if owner is not set. And `Owns` then takes care of it once the owner is set.


```
// MachineSetToDeployments is a handler.ToRequestsFunc to be used to enqueue requests for reconciliation
// for MachineDeployments that might adopt an orphaned MachineSet.
func (r *Reconciler) MachineSetToDeployments(ctx context.Context, o client.Object) []ctrl.Request {
```

Maybe we should add the comment here in the builder as well? (and of course keep Owns)

Reply:
@adityabhatia WDYT?

Reply:
Sry somehow lost track of this PR, I added comments now.